cmd/cosign/cli/verify/verify_bundle.go

```diff
@@ -149,15 +162,156 @@ func verifyNewBundle(ctx context.Context, bundlePath, trustedRootPath, keyRef, s
 	// Perform verification
 	payload, err := payloadBytes(artifactRef)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	buf := bytes.NewBuffer(payload)
 
 	sev, err := verify.NewSignedEntityVerifier(trustedmaterial, verifierConfig...)
 	if err != nil {
-		return err
+		return nil, err
+	}
+
+	return sev.Verify(bundle, verify.NewPolicy(verify.WithArtifact(buf), identityPolicies...))
+}
+
+func assembleNewBundle(ctx context.Context, sigBytes, signedTimestamp []byte, envelope *dsse.Envelope, artifactRef string, cert *x509.Certificate, ignoreTlog bool, sigVerifier signature.Verifier, pkOpts []signature.PublicKeyOption, rekorClient *client.Rekor) (*sgbundle.Bundle, error) {
```

Comment:
I would really like to see tests for this. I would take a look at the tests in `verify_blob.go` -- I authored the `keylessStack` test helper there (it was bundled into an advisory so my name is not in `git-blame`) which lets you do integration-y tests without running the e2e tests. This func should be covered if we assemble a trusted root there and try to verify some blobs.

Reply:
I added a happy-path test for `assembleNewBundle` and `verifyNewBundle`.